client/network/src/config.rs

```diff
@@ -95,6 +95,19 @@ pub struct Params<B: BlockT, H: ExHashT> {
 
 	/// Registry for recording prometheus metrics to.
 	pub metrics_registry: Option<Registry>,
+
+	/// Request response configuration for the grandpa warp sync request protocol.
+	///
+	/// [`RequestResponseConfig`] [`name`] is used to tag outgoing grandpa warp sync requests with
+	/// the correct protocol name. In addition all of [`RequestResponseConfig`] is used to handle
+	/// incoming grandpa warp sync requests, if enabled.
+	///
+	/// Can be constructed either via
+	/// [`grandpa_warp_sync_request_handler::generate_protocol_config`] allowing outgoing but not
+	/// incoming requests, or constructed via
+	/// [`grandpa_warp_sync_request_handler::GrandpaWarpSyncRequestHandler::new`] allowing both
+	/// outgoing and incoming requests.
+	pub grandpa_warp_sync_request_protocol_config: RequestResponseConfig,
```

Comment:
Considering that the actual protocol handling is not inside of `sc-network` but in a separate crate, I'd suggest to remove this field and instead ask the user to put the protocol in the `request_response_protocols` field, which is more generic. After all, nothing inside of `sc-network` distinguishes `NetworkConfiguration::grandpa_warp_sync_request_protocol_config` from `NetworkConfiguration::request_response_protocols`.